### PR TITLE
GEOMESA-1072 Invalid types in SFT parsing are not handled well

### PR DESCRIPTION
**Pull request by aheyne**
Added error handling for invalid SFT type with information on accepted values.

Signed-off-by: Austin Heyne <aheyne@ccri.com>

### CODE REVIEW

File: geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/SimpleFeatureTypes.scala [hunk 1]
```diff
@@ -102,6 +102,10 @@ object SimpleFeatureTypes {
 
     case t if specParser.parse(specParser.listType, t).successful => ListAttributeSpec(conf)
     case t if specParser.parse(specParser.mapType, t).successful  => MapAttributeSpec(conf)
+
+    case default => throw new IllegalArgumentException(s"Type $default is not a supported Simple Feature attribute type, "
+      + s"valid types are: ${simpleTypeMap.keys.toList.sorted}, ${geometryTypeMap.keys.toList.sorted}, "
+      + "or specs of the type \"id:Integer:opt1=v1:opt2=v2,*geom:Geometry:srid=4326,ct:List[String]:index=true,mt:Map[String,Double]:index=false\"")
```
Comment (elahrvivaz) [inline geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/SimpleFeatureTypes.scala]: could we change this last line to just be `List[?], Map[?,?]` ? that seems more readable to me.

File: geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/SimpleFeatureTypes.scala [hunk 2]
```diff
@@ -136,7 +140,6 @@ object SimpleFeatureTypes {
     }
     val defaultGeom = geomAttributes.find(_.options.get(OPT_DEFAULT).exists(_.toBoolean))
         .orElse(geomAttributes.headOption)
-    // TODO GEOMESA-594 allow for setting default date field
```
Comment (elahrvivaz) [inline geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/SimpleFeatureTypes.scala]: thanks for catching this

File: geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/SimpleFeatureTypes.scala (hunk 1 above)
Comment (aheyne) [inline geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/SimpleFeatureTypes.scala]: Does that change work?

**Comment by elahrvivaz** [comment]
:shipit:

**Pull request merged**